src/networkmanager/cni.hpp

```diff
@@ -51,28 +54,78 @@ class CNI : public CNIItf {
      * @param rt Runtime configuration parameters.
      * @return Error.
      */
-    Error DeleteNetworkList(const NetworkConfigList& net, const RuntimeConf& rt) override
-    {
-        (void)net;
-        (void)rt;
-
-        return ErrorEnum::eNone;
-    }
+    Error DeleteNetworkList(const NetworkConfigList& net, const RuntimeConf& rt) override;
 
     /**
      * Checks that a configuration is reasonably valid.
      *
      * @param net List of network configurations.
      * @return Error.
      */
-    Error ValidateNetworkList(const NetworkConfigList& net) override
-    {
-        (void)net;
+    Error ValidateNetworkList(const NetworkConfigList& net) override;
 
-        return ErrorEnum::eNone;
-    }
-};
+    /**
+     * Returns network list from cache.
+     *
+     * @param[out] net Network list.
+     * @param[out] rt Runtime configuration.
+     * @return Error.
+     */
+    Error GetNetworkListCachedConfig(NetworkConfigList& net, RuntimeConf& rt) override;
+
+private:
+    enum class Action { eAdd, eDel, eCheck };
```

Comment:
Use Aos enum, it allows converting string to enum and back. So, no need to `mActionsMap`.

Reply:
Done